src/widget/widgetengine.js

```diff
@@ -34,6 +35,22 @@ export default class WidgetEngine extends Plugin {
 			}
 
 			const viewSelection = conversionApi.viewSelection;
+
+			// Add CSS class if selection is placed inside nested editable that belongs to widget.
+			const editableElement = viewSelection.editableElement;
+
+			if ( editableElement && !( editableElement instanceof RootEditable ) ) {
+				const widget = editableElement.findAncestor( element => isWidget( element ) );
+
+				if ( widget ) {
+					widget.addClass( WIDGET_SELECTED_CLASS_NAME );
```

Comment:
Do we use the `I_AM_CONST` notation? I don't remember now what we've decided. cc @pjasiun, @scofalik